parcels/particleset.py

```diff
@@ -862,15 +709,11 @@ def set_variable_write_status(self, var, write_status):
 
     def execute(
         self,
+        endtime: timedelta | datetime,
+        dt: np.float64 | np.float32 | timedelta,
         pyfunc=AdvectionRK4,
-        pyfunc_inter=None,
-        endtime=None,
-        runtime: float | timedelta | np.timedelta64 | None = None,
-        dt: float | timedelta | np.timedelta64 = 1.0,
```

Comment:
Would we be able to update this to the following?
```python
    def execute(
        self,
        pyfunc=AdvectionRK4,
        *,
        endtime: timedelta | datetime | None = None,
        runtime = None, # Add a `if runtime is not None: raise NotImplementedError` in the code below
        dt: np.float64 | np.float32 | timedelta | None = None,
        output_file=None,
        verbose_progress=True,
    ):
```

This:
1. keeps the relative order of the args somewhat
2. forces everything after `*` to be a keyword only argument
3. flags intention for near-future dev that `runtime` will need to be implemented

I think (1) is important since users in v3 would be tripped up a bit by the position of `pyfunc` otherwise